compiler/rustc_resolve/messages.ftl

```diff
@@ -257,8 +257,11 @@ resolve_lowercase_self =
     attempt to use a non-constant value in a constant
     .suggestion = try using `Self`
 
+resolve_macro_cannot_use_as_attr = `macro_rules!` macros cannot be used as attribute macros
+resolve_macro_cannot_use_as_derive = `macro_rules!` macros cannot be used as derive macros
```

Comment:
This doesn't account for declarative macros 2.0 (`macro … { … }` under feature `decl_macro`). These diagnostics now call them macro_rules-macros (declarative macros 1.2).

It would be great if you could consider them (even if they're stuck in limbo atm ^^', cc https://youtu.be/r0H8BUiekcM).

To avoid the combinatorial explosion of {macro_rules,macro}{attribute,derive}, you could just go for *declarative macros cannot …* I guess. Alternatively, a static str in the subdiagnostic struct.

You should be able to obtain this information via `self.get_macro_by_def_id(macro_def_id).macro_rules`.